src/functions-and-arrays.js

```diff
@@ -1,73 +1,488 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else if (num1 < num2) {
+    return num2;
+  } else {
+    return num1, num2;
+  }
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(array) {
+  if (array.length === 0) {
+    return null;
+  } else if (array.length == 1) {
+    return array[0];
+  } else {
+    let longestWord = "";
+
+    for (let i = 0; i < array.length; i++) {
+      if (array[i].length > longestWord.length) {
+        longestWord = array[i];
+      }
+    }
+    return longestWord;
+  }
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(array) {
+  let sum = 0;
+  if (array.length === 0) {
+    return 0;
+  } else if (array.length === 1) {
+    sum += array[0];
+    return sum;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      if (array[i] === 0) {
+        sum = 0;
+      } else {
+        sum += array[i];
+      }
+    }
+    return sum;
+  }
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(array) {
+  let average = 0;
+  let sum = 0;
+  if (array.length == 0) {
+    return null;
+  } else if (array.length === 1) {
+    average += array[0];
+    return average;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      sum += array[i];
+    }
+    average = sum / array.length;
+    return average;
+  }
+}
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
+];
+
+function averageWordLength(array) {
+  let average = 0;
+  let sum = 0;
+  if (array.length === 0) {
+    return null;
+  } else if (array.length === 1) {
+    average = array[0].length;
+    return average;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      sum += array[i].length;
+    }
+    average = sum / array.length;
+    return average;
+  }
+}
 
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(array) {
+  if (array.length === 0) {
+    return null;
+  } else {
+    let newArray = [];
+    for (let i = 0; array.indexOf(array[i]) != -1; i++) {
```

Comment:
This was a good attempt to solve it! creating an empty array outside of the for loop and pushing elementsinto it!
Couple of  pointers:
1 - the condition in the for loop is a little messy, I understand the logic trying to avoid repeated words but doing a regular loop and working with the results inside the for loop would be better approach.
2 - For example. Inside the for loop you could create an if statement that only add the current string ( array[i] ) to the newArray IF that same string is not currently included in newArray. Using the .indexOf method.